src/ParameterLists.f90

```diff
@@ -2031,20 +2039,158 @@ SUBROUTINE getString_ParamType_scalar(thisParam,name,string,sskfmt,sdkfmt)
+          TYPE IS(ParamType_SDK_a2)
+            string=''
+            DO j=1,SIZE(param%val,DIM=2)
+              DO i=1,SIZE(param%val,DIM=1)
+                WRITE(tmpchar,TRIM(sdkfmtDef)) param%val(i,j)
+                string=string//TRIM(ADJUSTL(tmpchar))//delimDef
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SNK_a2)
+            string=''
+            DO j=1,SIZE(param%val,DIM=2)
+              DO i=1,SIZE(param%val,DIM=1)
+                string=string//str(param%val(i,j))//delimDef
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SLK_a2)
+            string=''
+            DO j=1,SIZE(param%val,DIM=2)
+              DO i=1,SIZE(param%val,DIM=1)
+                string=string//str(param%val(i,j))//delimDef
+              ENDDO
+            ENDDO
+!          TYPE IS(ParamType_SBK_a2)
+!            ALLOCATE(string(SIZE(param%val))
+!            DO i=1,SIZE(param%val)
+!              WRITE(tmpchar,'(L1)') param%val(i)
+!              string(i)=TRIM(ADJUSTL(tmpchar))
+!            ENDDO
+          TYPE IS(ParamType_STR_a2)
+            string=''
+            DO j=1,SIZE(param%val,DIM=2)
+              DO i=1,SIZE(param%val,DIM=1)
+                string=string//param%val(i,j)//delimDef
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SSK_a3)
+            string=''
+            DO k=1,SIZE(param%val,DIM=3)
+              DO j=1,SIZE(param%val,DIM=2)
+                DO i=1,SIZE(param%val,DIM=1)
+                  WRITE(tmpchar,TRIM(sskfmtDef)) param%val(i,j,k)
+                  string=string//TRIM(ADJUSTL(tmpchar))//delimDef
+                ENDDO
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SDK_a3)
+            string=''
+            DO k=1,SIZE(param%val,DIM=3)
+              DO j=1,SIZE(param%val,DIM=2)
+                DO i=1,SIZE(param%val,DIM=1)
+                  WRITE(tmpchar,TRIM(sdkfmtDef)) param%val(i,j,k)
+                  string=string//TRIM(ADJUSTL(tmpchar))//delimDef
+                ENDDO
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SNK_a3)
+            string=''
+            DO k=1,SIZE(param%val,DIM=3)
+              DO j=1,SIZE(param%val,DIM=2)
+                DO i=1,SIZE(param%val,DIM=1)
+                  string=string//str(param%val(i,j,k))//delimDef
+                ENDDO
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SLK_a3)
+            string=''
+            DO k=1,SIZE(param%val,DIM=3)
+              DO j=1,SIZE(param%val,DIM=2)
+                DO i=1,SIZE(param%val,DIM=1)
+                  string=string//str(param%val(i,j,k))//delimDef
+                ENDDO
+              ENDDO
+            ENDDO
+!          TYPE IS(ParamType_SBK_a3)
+!            ALLOCATE(string(SIZE(param%val))
+!            DO i=1,SIZE(param%val)
+!              WRITE(tmpchar,'(L1)') param%val(i)
+!              string(i)=TRIM(ADJUSTL(tmpchar))
+!            ENDDO
+!          TYPE IS(ParamType_STR_a3)
```

Comment:
Why is this commented out?

Reply:
The 3-D string arrays are commented out because there is no parameter list type for that dimensionality. In the event that parameter type gets added, this code can be uncommented. Still want to keep it?

Reply:
same here.